File: batch850/RestapiExampl850/app/src/main/java/com/example/restapiexampl850/ApiServices.java
package com.example.restapiexampl850;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ApiServices {
    @GET("posts")
    public Call<ArrayList<PostModel>> getalldata();

   @GET("post/{id}/comments")
   Call<ArrayList<Comments>> getcomments(
           @Path("id") int id
   );

   @FormUrlEncoded
   @POST("posts")
    Call<PostModel> getposts(
            @Field("userId") int userid,
            @Field("title") String title,
            @Field("body") String body
   );
}

File: Rahoul3pm/My Application/app/src/main/java/com/example/myapplication/ApiServices.kt
package com.example.myapplication


import com.example.myapplication.usermodels.UserModel
import retrofit2.Call
import retrofit2.http.*

interface ApiServices {

    @GET("posts")
    fun getpostlist() : Call<ArrayList<PostModel>>

   @GET("posts/{id}")
   fun  getpostwithid(
       @Path("id") id:Int
   ): Call<PostModel>


   @GET("comments")
   fun getcommentquery(
       @Query("postId") id:Int
   ): Call<ArrayList<CommentModel>>


   @FormUrlEncoded
   @POST("posts")
   fun createresourc(
       @Field("title") title:String,
       @Field("userId")  userId:Int,
       @Field("body") body:String
   ): Call<PostModel>

   @GET("users")
   fun getusers(): Call<ArrayList<UserModel>>

   @GET("photos")
   fun getphotos() : Call<ArrayList<ImageModel>>
}